hera_stats/jackknives.py

```diff
@@ -0,0 +1,402 @@
+        uvp_avg.set_stats("bootstrap_errs", (spw, uvp.get_blpairs()[0], pol), errs[None])
+
+    return uvp_avg
+
+def bootstrap_jackknife(uvpl, pol="xx", n_boots=50):
+    """
+    Bootstraps a list of jackknife pairs.
+
+    Parameters
+    ----------
+    uvpl: list of UVPSpec pairs
+        The jackknived data, with UVPSpec.labels as the group and UVPSpec.jkftype as 
+        the jackknife type, shape=(n, 2).
+
+    pol: string, optional
+        Polarization to use for bootstrapping. Default="xx".
+
+    n_boots: int, optional
+        Number of times to bootstrap to create a sample. Default: 50.
+    """
+    data = []
+    for uv in uvpl:
+        bspair = []
+        for u in uv:
+            uc = copy.deepcopy(u)
+            bspair.append(_bootstrap_uvp_once(uc, pol=pol, n_boots=n_boots))
+        data.append(bspair)
+    return data
+
+def save_jackknife(uvpl, savename=None):
+    """
+    Saves a bootstrapped jackknife pair list.
+
+    uvpl: list of UVPSpecs
+        The list to save. Note: will save jackknives with no bootstrap,
+        so make sure to run uvpl through bootstrap_jackknife first.
+
+    savename: string, optional
+        Outfile name. Will append with a bunch of information.
+        Default: None
+    """
+    if not os.path.exists("./data"):
+        os.mkdir("./data")
+
+    # Check if all jackknives are the same type
+    jkftypes = [u.jkftype  for uvp in uvpl for u in uvp]
+    if all([j == jkftypes[0] for j in jkftypes]):
+        jkftype = jkftypes[0]
+    else:
+        raise AttributeError("All jackknifes must be of the same type")
+
+    fname = "jackknife." + jkftype + ".Nj%i." % len(uvpl) + utils.timestamp()
+    if savename is not None:
+        fname = savename + "." + fname
+
+    # Open pspec container
+    pc = hp.PSpecContainer("./data/" + fname, "rw")
+
+    for i, uvp_pair in enumerate(uvpl):
+        jkf = "%s.%i" % (uvp_pair[0].jkftype, i)
+        for k, uvp in enumerate(uvp_pair):
+            # Save pspec to group {jackknife_type}.{jackknife_num} and 
+            # pspec grp{groupnumber}
+            name = "grp%i" % k
+            pc.set_pspec(jkf, psname=name, pspec=uvp)
+
+def split_ants(uvp, n_jacks=40, verbose=False):
+    """
+    Splits available antenna into two groups randomly, and returns the
+    UVPspec of each.
+
+    Parameters
+    ----------
+    uvp: list or single UVPSpec
+        The data to use for jackkniving.
+
+    n_jacks: int, optional
+        Number of times to jackknife the data. Default: 40.
+
+    verbose: boolean, optional
+        If true, prints actions.
+
+    Returns
+    -------
+    uvpl: list
+        List of hera_pspec.UVPSpecData objects that have been split
+        accordingly.
+    """
+    if isinstance(uvp, list):
+        uv = uvp[0]
+        for u in uvp[1:]:
+            uv += u
+        uvp = uv
+
+    # Load all baselines in uvp
+    blns = [uvp.bl_to_antnums(bl) for bl in uvp.bl_array]
```

Comment:
Need to check that UVPSpec only contains a single redundant group? Or ask the user to specify which redundant group they care about?